fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -1589,6 +1605,168 @@ private static QueryParameter parseChainedParameter(Class<?> resourceType, Strin
+
+            for (String component : components) {
+                // Split into subcomponents by colon delimiter
+                List<String> subcomponents = Arrays.asList(component.split(SearchConstants.COLON_DELIMITER_STR, 3));
+
+                // Validate correct number of subcomponents
+                if ((currentIndex < lastIndex && subcomponents.size() != 2) ||
+                        (currentIndex == lastIndex && subcomponents.size() != 3)) {
+                    throw SearchExceptionUtil.buildNewInvalidSearchException(INCORRECT_NUMBER_OF_COMPONENTS_FOR_REVERSE_CHAIN_SEARCH);
+                }
+
+                // Validate referenced-by resource type
+                String referencedByResourceTypeName = subcomponents.get(0);
+                Class<? extends Resource> referencedByResourceType = ModelSupport.getResourceType(referencedByResourceTypeName);
+                if (referencedByResourceType == null) {
+                    throw SearchExceptionUtil.buildNewInvalidSearchException(
+                        String.format(INVALID_RESOURCE_TYPE_FOR_REVERSE_CHAIN_SEARCH, referencedByResourceTypeName));
+                }
+
+                // Validate reference search parameter
+                String referenceSearchParameterName = subcomponents.get(1);
+                SearchParameter referenceSearchParameter = getSearchParameter(referencedByResourceType, referenceSearchParameterName);
+                if (referenceSearchParameter == null) {
+                    throw SearchExceptionUtil.buildNewInvalidSearchException(
+                        String.format(SEARCH_PARAMETER_NOT_FOUND, referenceSearchParameterName, referencedByResourceTypeName));
+                }
+                if (!Type.REFERENCE.equals(Type.fromValue(referenceSearchParameter.getType().getValue()))) {
+                    throw SearchExceptionUtil.buildNewInvalidSearchException(
+                        String.format(PARAMETER_TYPE_NOT_REFERENCE_FOR_REVERSE_CHAIN_SEARCH, referenceSearchParameterName));
+                }
+
+                // Validate resource type is one of the reference search parameter target resource types
+                if (!referenceSearchParameter.getTarget().contains(ResourceType.of(resourceType.getSimpleName()))) {
+                    throw SearchExceptionUtil.buildNewInvalidSearchException(
+                        String.format(TARGET_TYPE_OF_REFERENCE_PARAMETER_NOT_VALID_FOR_REVERSE_CHAIN_SEARCH,
+                            referenceSearchParameterName, resourceType.getSimpleName()));
+                }
+
+                // Create new QueryParameter
+                QueryParameter parameter = new QueryParameter(Type.REFERENCE, referenceSearchParameterName, Modifier.TYPE, referencedByResourceTypeName, false, true);
+                if (rootParameter == null) {
+                    rootParameter = parameter;
+                } else {
+                    if (rootParameter.getChain().isEmpty()) {
```

Comment:
```suggestion
                } else if (rootParameter.getChain().isEmpty()) {
```

Reply:
The Else if / Else seems like it could be reduced one level

Reply:
done